plugins/amdgpu/tests/test_topology_remap.c

```diff
@@ -0,0 +1,985 @@
+/**************************************************************************************************
+ * GPU groups remapping unit tests
+ *
+ * Test cases for GPU topology group remapping when there are P2P iolinks between the GPUs. GPUs are
+ * considered to be grouped when they have a P2P iolink.
+ * 	a. P2P-XGMI: They are connected through a XGMI bridge
+ * 	b. P2P-PCIe: They have large BAR enabled and each GPU can address full address space of the
+ *         other GPU (44-bits address limitation)
+ *
+ * When GPUs have large BAR, but each GPU cannot address full address of each other, they do not
+ * have P2P-PCIe links and are not considered for GPU grouping. The GPU still has to be remapped to
+ * a GPU with large BAR on restore.
+ *
+ * GPUs can be part of 2 groups at the same time, i.e have P2P-XGMI links with one set of GPUs and
+ * have P2P-PCIe links with another set of GPUs.
```

Comment:
This is an old comment and I forgot to remove it.

Initially, I was planning to implement it the way you mentioned (G1, G5, G6, G7, G8) but when I saw that each GPU would actually be part of 5 groups instead of 2, I thought it would make things too complicated. 
For now, it seems that the current code (only considering XGMI groups) can handle all uses cases ok, so I left it as is. I think we may have to re-visit this when we start looking at p2p_links with the dkms branch.